eureka-client/src/main/java/com/netflix/discovery/shared/Application.java

```diff
@@ -216,14 +216,15 @@ private void _shuffleAndStoreInstances(boolean filterUpInstances, boolean indexB
         synchronized (instances) {
             instanceInfoList = new ArrayList<InstanceInfo>(instances);
         }
-        if (indexByRemoteRegions || filterUpInstances) {
+        boolean remoteIndexingActive = indexByRemoteRegions && null != instanceRegionChecker && null != clientConfig
```

Comment:
code will iterate and filter whether remote indexing is active or not; here's the guard condition:
if (remoteIndexingActive OR filterUpInstances) {
 if filterUpInstances [do filtering]
 if remoteIndexingActive [do remote indexing]
}

.